src/main.rs

```diff
@@ -0,0 +1,230 @@
+
+    println!("Enter the name of raw_food x : ");
+    let x: String = read_input()?;
+    let q = format!(
+        "match $rf isa raw_food, has name \"{x}\";
+        $is_i (raw_food: $rf,dish: $d) isa is_ingredient;
+        $sl (seller: $m,product: $d) isa sells;
+        $m has name $n;
+        get $n;"
+    );
+    let mut answer_stream = transaction
+        .query()
+        .match_(q.as_str())
+        .map_err(HandleError::TypeDB)?;
+    while let Some(result) = answer_stream.next().await {
+        match result {
+            Ok(concept_map) => {
+                for (_, concept) in concept_map {
+                    if let Concept::Attribute(Attribute {
+                        value: Value::String(value),
+                        ..
+                    }) = concept
+                    {
+                        println!("{}", value);
+                    }
+                }
+            }
+            Err(err) => {
+                return Err(HandleError::TypeDB(err));
+            }
+        }
+    }
+    Ok(())
+}
+
+async fn get_strange_menu(connection: Connection) -> Result<(), HandleError> {
+    let databases = DatabaseManager::new(connection.clone());
+    let session = Session::new(
+        databases
+            .get(MENU_DATABASE)
+            .await
+            .map_err(HandleError::TypeDB)?,
+        Data,
+    )
+    .await
+    .unwrap();
+    let transaction = session
+        .transaction(Read)
+        .await
+        .map_err(HandleError::TypeDB)?;
+
+    println!("::Q2::");
+    let q = "
+    match $m isa menu, has is_vegetarian false,has name $n;
+    $d isa dish, has is_vegetarian false;
+    $sp (restaurant: $m,$d) isa speciality;
+    get $m;count;"
+        .to_string();
+    let answer = transaction
+        .query()
+        .match_aggregate(q.as_str())
+        .await
+        .map_err(HandleError::TypeDB)?;
+    println!("Answe for Q2:  {}\n", answer.into_i64());
+    Ok(())
+}
+
+async fn get_raw_items(connection: Connection) -> Result<(), HandleError> {
+    let databases = DatabaseManager::new(connection.clone());
+    let session = Session::new(
+        databases
+            .get(MENU_DATABASE)
+            .await
+            .map_err(HandleError::TypeDB)?,
+        Data,
+    )
+    .await
+    .unwrap();
+    let transaction = session
+        .transaction(Read)
+        .await
+        .map_err(HandleError::TypeDB)?;
+
+    println!("Enter the avg_rating of Restraunt r : ");
+    let avg_rating: String = read_input()?;
+    println!("Enter the threshold price for raw_ingredient p : ");
+    let price: String = read_input()?;
+
+    println!("::Q3::");
+    let q = format!(
+        "match $m isa menu, has avg_rating>{avg_rating},has name $mn;
+        $d isa dish, has name $dn;
+        $sl (seller: $m,product: $d) isa sells, has price $p;
+        $p>{price};
+        $rf isa raw_food,has name $rfn;
+        $isn (raw_food: $rf,$d) isa is_ingredient;
+        get $rfn;"
```

Comment:
This query gets the names of raw foods used as ingredients in dishes above a certain price at places with ratings higher than a certain rating. I'd remove the part about counting from your description in `query_runner` and not modify this query as we already use count in our second query.